src/test/java/uk/gov/ons/ctp/integration/contcencucumber/cucSteps/sso/TestSSOFieldLauncherService.java

```diff
@@ -0,0 +1,325 @@
+		String localTextToFind = "/session?token";
+		assertTrue(currentURL.contains(devTextToFind) || currentURL.contains(localTextToFind));
+    }
+    
+    @Given("that the response to a CCS interview job has already been submitted")
+    public void that_the_response_to_a_CCS_interview_job_has_already_been_submitted() {
+        log.info("change the base URL to be one for an interview that has already been submitted");
+        baseUrl = completedUrl;
+    }
+    
+    @Then("the completion message {string} is displayed to me")
+    public void the_completion_message_is_displayed_to_me(String completionMessage) {
+    	
+    	String titleText = questionnaireCompleted.getCCSCompletedTitleText();
+    	assertEquals("CCS Completion title has incorrect text", completionMessage, titleText);
+    }
+
+    @Given("I have already entered my credentials for SSO")
+    public void i_have_already_entered_my_credentials_for_SSO() {
+    	log.with(baseUrl).debug("The job URL that was clicked on");
+    	driver.get(baseUrl);
+    	
+    	try {
+			log.info("Sleep for 5 seconds to give the SSO page time to appear");
+			Thread.sleep(5000);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+    	
+    	log.with(userId).debug("Now entering my credentials for the SSO");
+        sso.enterUserId(userId);
+        sso.clickNextButton();
+        sso.enterPassword(pw);
+        sso.clickSignInButton();
+    }
+    
+    @When("I click on the job URL⁠ in the chrome browser in a new window")
+    public void i_click_on_the_job_URL⁠_in_the_chrome_browser_in_a_new_window() {
+    	// Store the current window handle
+    	String winHandleBefore = driver.getWindowHandle();
+
+    	// Perform the click operation that opens new window
+    	JavascriptExecutor jse = (JavascriptExecutor)driver;
+    	jse.executeScript("window.open()");
+    	ArrayList<String> tabs = new ArrayList<String>(driver.getWindowHandles());
+    	driver.switchTo().window(tabs.get(1));
+    	
+    	try {
+			log.info("Sleep for 5 seconds to give the new tab time to appear");
+			Thread.sleep(5000);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+    	
+        driver.get(baseUrl);
+    }
+    
+    @Then("I am not presented with the SSO screen to enter my credentials")
+    public void i_am_not_presented_with_the_SSO_screen_to_enter_my_credentials() {
+        
+    	try {
+			log.info("Sleep for 5 seconds to give the SSO page time to appear");
+			Thread.sleep(5000);
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		}
+ 
+    	String titleText = "";
+		try {
+			titleText = sso.getSSOTitleText();
+		} catch (NoSuchElementException e) {
+			log.info("We are expecting that the SSO screen will not appear, therefore there should not be any SSO title element found");
+		}
+		
+    	log.with(titleText).debug("The SSO title text if found");
+    	assertNotEquals("SSO title should not have appeared", "Sign in with your Google Account", titleText);
```

Comment:
This fail-if-present logic would be clearer if it was right after 'sso.getSSOTitleText()'. I assume that if the code can hold of the SSO title then it should fail the test, and if so then it's a straightforward fail with no comparisons or anything.

Reply:
Yes I agree, that sounds like a better way to do it.